packages/react-utils/src/rewriteReactNodeRecursively.ts

```diff
@@ -0,0 +1,37 @@
+import { ReactNode, isValidElement, cloneElement, Key } from 'react';
+
+type RewriteReactNodeRecursivelyCallback = (stringSubNode: string) => string;
+
+/**
+ * Recursively crawls ReactNode and applies callback on all string subnodes
+ * @param node ReactNode to be recursively crawled & modified
+ * @param callback function to be applied on all string subnodes
+ * @param reactKey optional React key applied on the root node, if it is a ReactElement. Used internally.
+ * @returns modified ReactNode
+ */
+export const rewriteReactNodeRecursively = (
+    node: ReactNode,
+    callback: RewriteReactNodeRecursivelyCallback,
+    reactKey?: Key,
+): ReactNode => {
+    if (typeof node === 'string') return callback(node);
+    if (typeof node === 'number') return callback(node.toString());
+
+    if (Array.isArray(node)) {
+        return node.map((child, index) => {
+            const newReactKey: Key | undefined = isValidElement(child)
+                ? child.key ?? index
```

Comment:
When rewriting the `ReactNode`, I **do** have to provide key even when the original element did not.
This is not an edge case, it's actually very common:
`<span>label<i>123</i></span>`
the `span` has children = array `["label", <i>123</i>]`. Now the `<i>` has to have a `key`, idk why :facepalm: